backend/convert_to_protobuf.go

```diff
@@ -108,19 +109,30 @@ func (t convertToProtobuf) QueryDataRequest(req *QueryDataRequest) *pluginv2.Que
 }
 
 func (t convertToProtobuf) QueryDataResponse(res *QueryDataResponse) (*pluginv2.QueryDataResponse, error) {
-	encodedFrames := make([][]byte, len(res.Frames))
-	var err error
-	for i, frame := range res.Frames {
-		encodedFrames[i], err = data.MarshalArrow(frame)
+	pQDR := &pluginv2.QueryDataResponse{
+		Responses: make([]*pluginv2.DataResponse, len(res.Responses)),
+		Metadata:  res.Metadata,
+	}
+	for rIdx, dr := range res.Responses {
+		encodedFrames, err := data.FramesToBytesSlice(dr.Frames)
 		if err != nil {
 			return nil, err
 		}
+		pQDR.Responses[rIdx] = &pluginv2.DataResponse{
+			RefId:  dr.RefID,
+			Frames: encodedFrames,
+		}
+
+		b, err := json.Marshal(dr.Meta.Custom)
```

Comment:
This isn't directly related to the issue this is on. Just something I want to try, and see if we can marshall and unmarhsal this object for the plugin author, and they can assert that it is of their type. 

I don't think will allow for any custom marshalling however. so maybe the bytes pattern is better.